v4l2loopback.c

```diff
@@ -2369,7 +2369,7 @@ static int v4l2_loopback_add(struct v4l2_loopback_config *conf, int *ret_nr)
 	int _max_height = DEFAULT_FROM_CONF(
 		max_height, < V4L2LOOPBACK_SIZE_MIN_HEIGHT, max_height);
 	bool _announce_all_caps = (conf && conf->announce_all_caps >= 0) ?
-					  (conf->announce_all_caps) :
+					  (!!conf->announce_all_caps) :
```

Comment:
which problem does this solve?

Reply:
Mostly a correctness fix to ensure unexpected values like `42` don't propagate into the internal state and are normalized to boolean `0` and `1` instead. Avoids unexpected behavior in places that assume the internal state to be exactly `0` or `1`.